### PR TITLE
Converting gear tweaks to an overridable proc.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
This should have no player-facing changes but allows downstream or modpacks to override gear tweaks.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Bugging out, not sure why.

File: code/modules/client/preference_setup/loadout/lists/accessories.dm [hunk 1]
```diff
@@ -10,29 +10,34 @@
 
 /datum/gear/accessory/tie/New()
 	..()
```
Comment (Techhead0) [inline code/modules/client/preference_setup/loadout/lists/accessories.dm]: Extraneous new

File: mods/corporate/datum/loadout.dm [hunk 2]
```diff
@@ -73,13 +75,12 @@
 	display_name = "corporate jacket selection"
 	path = /obj/item/clothing/suit
 
-/datum/gear/suit/corp_jacket/New()
-	..()
-	var/jackets = list(
+/datum/gear/suit/corp_jacket/get_gear_tweak_options()
+	. = ..()
+	.[/datum/gear_tweak/path/specified_types_list] = list(
```
Comment (Andrew-Fall) [inline mods/corporate/datum/loadout.dm]: Lazyinit

File: code/modules/client/preference_setup/loadout/lists/utility.dm [hunk 3]
```diff
@@ -84,9 +85,8 @@ modular computers
 	path = /obj/item/modular_computer/tablet
 	cost = 4
 
-/datum/gear/utility/customtablet/New()
-	..()
-	gear_tweaks += new /datum/gear_tweak/tablet()
+/datum/gear/utility/customtablet/get_gear_tweak_options()
+	. = ..() | /datum/gear_tweak/tablet
```
Comment (Andrew-Fall) [inline code/modules/client/preference_setup/loadout/lists/utility.dm:89]: Perhaps Lazyinit .[/datum/gear_tweak/tablet] for consistency

Comment (MistakeNot4892) [reply code/modules/client/preference_setup/loadout/lists/utility.dm:89]: It needs to be non-associative if I want it to be 1:1 with previous implementation.

**Comment by MistakeNot4892** [comment]
All works locally, ready for review.

**Review by afterthought2 — APPROVED** [review]

**Review by deathride58 — APPROVED** [review]
Should be good for speedmerge

**Review by Andrew-Fall — APPROVED** [review]

**Pull request merged**